presentation/src/main/java/com/anytypeio/anytype/presentation/relations/option/CreateOrEditOptionViewModel.kt

```diff
@@ -25,7 +33,133 @@ class CreateOrEditOptionViewModel(
 ) : BaseViewModel() {
 
     val command = MutableSharedFlow<Command>(replay = 0)
-    val viewState: MutableStateFlow<OptionScreenViewState?>? = null
+    val viewState: MutableStateFlow<CreateOrEditOptionScreenViewState> =
+        MutableStateFlow(initialViewState())
+
+    private fun initialViewState(): CreateOrEditOptionScreenViewState {
+        val optionId = viewModelParams.optionId
+        val color = getOptionColor()
+        return if (optionId != null) {
+            Timber.d("Editing option with id: $optionId")
+            CreateOrEditOptionScreenViewState.Edit(
+                optionId = optionId,
+                text = viewModelParams.name.orEmpty(),
+                color = color
+            )
+
+        } else {
+            Timber.d("Creating new option")
+            CreateOrEditOptionScreenViewState.Create(
+                text = viewModelParams.name.orEmpty(),
+                color = color
+            )
+        }
+    }
+
+    fun updateName(name: String) {
+        viewState.value = when (val state = viewState.value) {
+            is CreateOrEditOptionScreenViewState.Create -> state.copy(text = name)
+            is CreateOrEditOptionScreenViewState.Edit -> state.copy(text = name)
+        }
+    }
+
+    fun updateColor(color: ThemeColor) {
+        viewState.value = when (val state = viewState.value) {
+            is CreateOrEditOptionScreenViewState.Create -> state.copy(color = color)
+            is CreateOrEditOptionScreenViewState.Edit -> state.copy(color = color)
+        }
+    }
+
+    fun onButtonClick() {
+        when (viewState.value) {
+            is CreateOrEditOptionScreenViewState.Create -> createOption()
+            is CreateOrEditOptionScreenViewState.Edit -> updateOption()
+        }
+    }
+
+    private fun createOption() {
+        viewModelScope.launch {
+            val params = CreateRelationOption.Params(
+                space = spaceManager.get(),
+                relation = viewModelParams.relationKey,
+                name = viewState.value.text,
+                color = viewState.value.color.code
+            )
+            if (params.name.isNotEmpty()) {
```

Comment:
maybe show toast with "Please enter name for new option" in the `else` case?

Reply:
In a case of an empty title user can see a hint "Enter name..."